src/DurableSDK/PowerShellServices.cs

```diff
@@ -6,50 +6,140 @@
 namespace Microsoft.Azure.Functions.PowerShellWorker.Durable
 {
     using System;
+    using System.Collections.ObjectModel;
+    using System.Linq;
     using System.Management.Automation;
-    using PowerShell;
+    using Microsoft.Azure.Functions.PowerShellWorker.PowerShell;
+    using Microsoft.Azure.Functions.PowerShellWorker.Utility;
+    using Microsoft.Azure.WebJobs.Script.Grpc.Messages;
+    using Newtonsoft.Json;
 
     internal class PowerShellServices : IPowerShellServices
     {
-        private const string SetFunctionInvocationContextCommand =
-            "Microsoft.Azure.Functions.PowerShellWorker\\Set-FunctionInvocationContext";
+        private readonly string SetFunctionInvocationContextCommand;
 
         private readonly PowerShell _pwsh;
-        private bool _hasSetOrchestrationContext = false;
+        private bool hasInitializedDurableFunctions = false;
+        private readonly bool hasExternalDurableSDK = false;
 
         public PowerShellServices(PowerShell pwsh)
         {
-            _pwsh = pwsh;
+            // We attempt to import the external SDK upon construction of the PowerShellServices object.
+            // We maintain the boolean member hasExternalDurableSDK in this object rather than
+            // DurableController because the expected input and functionality of SetFunctionInvocationContextCommand
+            // may differ between the internal and external implementations.
+
+            try
+            {
+                pwsh.AddCommand(Utils.ImportModuleCmdletInfo)
```

Comment:
For this scenario, what if we did the following?

1) Do a `Get-Module -list PowerShellWorkerStrings.ExternalDurableSDKName` and log the module name and version. This will tell us if the module is the worker path. Additionally, you could track or build a detector which will tell you what `PowerShellWorkerStrings.ExternalDurableSDKName` version is currently being used by our customers. 
2) If the output of (1) is not null, then import the module
3) If the module could not be imported, get the error details using the `Get-Error` cmdlet and log it (to Kusto) and display the error `exceptionMessage` to the user. Follow up question: Are these customer logs, Kusto logs or both?

Reply:
Thanks @Francisco-Gamino. I've incorporated most of this feedback, please see my summary of latest changes here: https://github.com/Azure/azure-functions-powershell-worker/pull/839#issuecomment-1344568470